Modules/+Sources/Msquared.m

```diff
@@ -0,0 +1,481 @@
+            if val
+                try
+                    obj.tune(obj.setpoint_) % If this line is neglected, the laser will lock to whatever the current value is (which often is slightly off the setpoint).
+                catch
+                    obj.getFrequency();
+                end
+            else
+                obj.getFrequency();
+            end
+        end
+    end
+    
+    methods     % TunableLaser_invisible methods
+        function TuneCoarse(obj, target)
+            obj.TuneSetpoint(target);
+            obj.do_wavelength_lock = false;
+        end
+        function TunePercent(obj, target)
+            assert(target >= 0 && target <= 100, 'Target must be a percentage')
+            obj.resonator_percent = target;
+        end
+        function TuneSetpoint(obj, target)
+            obj.tune(obj.c/target);
+        end
+        function percent = GetPercent(obj)
+            obj.getFrequency();
+            obj.resonator_percent = obj.resonator_voltage/2;
+            percent = obj.resonator_percent;
+        end
+        function freq = getFrequency(obj)
+            obj.getWavemeterWavelength();
+            obj.callStatus();
+            wavelength = obj.determineResultingWavelength();
+            freq = obj.c/wavelength;    % Not sure if this should be used; imprecise.
+            obj.setpoint = freq;
+        end
+    end
+    
+    methods     % Module interpretation functions.
+        function module = determineModule(obj, wavelength)                  % Determines the module most appropriate for the target wavelength. If wavelength is not given, uses wavelength = obj.setpoint_
+            if nargin < 2
+                wavelength = obj.setpoint_;
+            end
+            
+            emm_crystal = 2;    % For now, the second (582-661 nm) crystal is the only one enabled. In the future, this should reference a pref.
+            
+            if     wavelength >= 515 && wavelength <= 582 && emm_crystal == 1   % For the 515-525 nm range where crystal 1 EMM and ECD-X have overlap, prefer EMM. There should be some way to override.
+                module = Sources.Msquared.moduleVIS;    %'EMM';
+            elseif wavelength >= 350 && wavelength <= 525
+                module = Sources.Msquared.moduleUV;     %'ECD-X';
+            elseif wavelength >= 515 && wavelength <= 582 && emm_crystal ~= 2
+                error('Crystal is currently hard-coded to 2, and EMM wavelengths in the 515-582 nm range are unavailible. Change crystal and/or remove hardcoding to fix.')
+            elseif wavelength >= 580 && wavelength <= 661 && emm_crystal == 2
+                module = Sources.Msquared.moduleVIS;    %'EMM';
+            elseif wavelength >= 700 && wavelength <= 1000
+                module = Sources.Msquared.moduleNIR;    %'SolsTiS';
+            else
+                module = '';    % Return empty if the wavelength is invalid. Easy to check with isempty().
+            end
```

Comment:
I don't have a super strong feeling about this, but could this maybe be stored as some dictionary that's a property of the class? would be useful for any laser with multiple ranges like this (thinking ahead to what the cwave might be like, though that's going to be an entirely different beast so probably will be entirely different code...)